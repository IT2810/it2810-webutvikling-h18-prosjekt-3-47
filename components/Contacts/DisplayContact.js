/**
 * DisplayContact displays the contact information, including phone number and address, of a contact.
 * The contact is sent as props to the component and the contact information is displayed in three text components.
 */
import React from 'react';
import { StyleSheet, Text, View } from 'react-native';
import c from '../../constants/Colors';

export default class DisplayContact extends React.Component {

    constructor(props) {
        super(props);
    }

    render() {
        return (
            <View style={styles.container}>
                <Text style={styles.text}>Name: {this.props.contact.name}</Text>
                <Text style={styles.text}>Telephone number: {this.props.contact.phoneNumber}</Text>
                <Text style={styles.text}>Address: {this.props.contact.address}</Text>
            </View>
        )
    }
}
const styles = StyleSheet.create({
    container: {
        flex: 1,
        backgroundColor: c.appMainBackground,
    },
    text: {
        marginLeft: '3%',
        fontSize: 16,
        marginTop: '1%',
    },
});
