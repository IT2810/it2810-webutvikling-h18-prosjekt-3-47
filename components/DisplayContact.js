import React from 'react';
import {
    StyleSheet,
    Text,
    View,
} from 'react-native';

export default class DisplayContact extends React.Component {

    constructor(props) {
        super(props);
    }

    render() {
        return (
            <View style={styles.container}>
                <Text style={styles.text}>Name: {this.props.contact.name}</Text>
                <Text style={styles.text}>Telephone number: {this.props.contact.phoneNumber}</Text>
                <Text style={styles.text}>Address: {this.props.contact.address}</Text>
            </View>
        )
    }
}
const styles = StyleSheet.create({
    container: {
        flex: 1,
    },
    text: {
        marginLeft: 12,
        fontSize: 16,
        marginTop: 6,
    },
});
