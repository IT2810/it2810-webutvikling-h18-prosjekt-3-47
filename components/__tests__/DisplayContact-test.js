import 'react-native';
import React from 'react';
import DisplayContact from '../Contacts/DisplayContact';
import renderer from 'react-test-renderer';
import ShallowRenderer from 'react-test-renderer/shallow';

let shallowRenderer = new ShallowRenderer();

let contacts = {
    contact: {
        name: "Navn Navnesen 1",
        phoneNumber: 12345678,
        address: "Adresse 1"
    }
};


it('renders correctly', () => {
    const tree = renderer.create(<DisplayContact contact={contacts.contact} />).toJSON();

    expect(tree).toMatchSnapshot();
});


it('rendering fails without props', () => {
    expect(() => {renderer.create(<DisplayContact/>)}).toThrow();

});

it('renders shallow correctly', () => {
    shallowRenderer.render(<DisplayContact contact={contacts.contact} />);
    const result = shallowRenderer.getRenderOutput();

    expect(result).toMatchSnapshot();
});
