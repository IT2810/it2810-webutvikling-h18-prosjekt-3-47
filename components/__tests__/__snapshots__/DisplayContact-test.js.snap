// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<View
  style={
    Object {
      "flex": 1,
    }
  }
>
  <Text
    accessible={true}
    allowFontScaling={true}
    ellipsizeMode="tail"
    style={
      Object {
        "fontSize": 16,
        "marginLeft": "3%",
        "marginTop": "1%",
      }
    }
  >
    Name: 
    Navn Navnesen 1
  </Text>
  <Text
    accessible={true}
    allowFontScaling={true}
    ellipsizeMode="tail"
    style={
      Object {
        "fontSize": 16,
        "marginLeft": "3%",
        "marginTop": "1%",
      }
    }
  >
    Telephone number: 
    12345678
  </Text>
  <Text
    accessible={true}
    allowFontScaling={true}
    ellipsizeMode="tail"
    style={
      Object {
        "fontSize": 16,
        "marginLeft": "3%",
        "marginTop": "1%",
      }
    }
  >
    Address: 
    Adresse 1
  </Text>
</View>
`;

exports[`renders shallow correctly 1`] = `
<Component
  style={
    Object {
      "flex": 1,
    }
  }
>
  <Text
    accessible={true}
    allowFontScaling={true}
    ellipsizeMode="tail"
    style={
      Object {
        "fontSize": 16,
        "marginLeft": "3%",
        "marginTop": "1%",
      }
    }
  >
    Name: 
    Navn Navnesen 1
  </Text>
  <Text
    accessible={true}
    allowFontScaling={true}
    ellipsizeMode="tail"
    style={
      Object {
        "fontSize": 16,
        "marginLeft": "3%",
        "marginTop": "1%",
      }
    }
  >
    Telephone number: 
    12345678
  </Text>
  <Text
    accessible={true}
    allowFontScaling={true}
    ellipsizeMode="tail"
    style={
      Object {
        "fontSize": 16,
        "marginLeft": "3%",
        "marginTop": "1%",
      }
    }
  >
    Address: 
    Adresse 1
  </Text>
</Component>
`;
